src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -422,6 +422,22 @@ public String getCookie() {
         }
 
         @Override public CauseOfBlockage getCauseOfBlockage() {
+            Run<?, ?> run = runForDisplay();
+            if (!stopping && run != null && !run.isLogUpdated()) {
+                stopping = true;
+                LOGGER.warning(() -> "Refusing to build " + this + " and cancelling it because associated build is complete");
+                Timer.get().execute(() -> {
+                    Queue.getInstance().cancel(this);
+                });
+            }
+            if (stopping) {
+                return new CauseOfBlockage() {
+                    @Override
+                    public String getShortDescription() {
+                        return "Stopping " + getDisplayName();
+                    }
+                };
```

Comment:
Not sure if an anonymous class or hard-coded text is ok here. My thought was that this cause should not usually be around long enough for anyone to see it, but I guess we should set up localization just in case.

Reply:
An anonymous class should be fine here. As you say, it ought not appear in the GUI for more than a moment if at all.